### PR TITLE
feat(content): Empty case for highlights

### PR DESCRIPTION
**Pull request by k88hudson**
I also added a couple of things to shim so we can see changes more easily

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/activity-streams/528)

### CODE REVIEW

File: content-src/components/Spotlight/Spotlight.js [hunk 1]
```diff
@@ -12,14 +12,15 @@ const SpotlightItem = React.createClass({
...
+      onDelete: function() {},
+      bestImage: {}
     };
   },
   render() {
     const site = this.props;
     const image = site.bestImage;
     const imageUrl = image.url;
-    const description = site.description;
+    const description = site.description || site.url;
```
Comment (pdehaan) [inline content-src/components/Spotlight/Spotlight.js]: Not sure if displaying URL here would be confusing when/if we do #407.

File: content-src/selectors/selectors.js [hunk 2]
```diff
@@ -7,26 +7,59 @@ const urlParse = require("url-parse");
 const SPOTLIGHT_LENGTH = module.exports.SPOTLIGHT_LENGTH = 3;
 const TOP_SITES_LENGTH = module.exports.TOP_SITES_LENGTH = 6;
 const BACKGROUND_FADE = 0.5;
-const DEFAULT_FAVICON_BG_COLOR = [255, 255, 255];
+const DEFAULT_FAVICON_BG_COLOR = [150, 150, 150];
```
Comment (pdehaan) [inline content-src/selectors/selectors.js]: Ha! Didn't we just change this to white 2 days ago in https://github.com/mozilla/activity-streams/pull/511/commits/4fec351d2e4cdd6ff69980820b5c380548742bef?

Comment (k88hudson) [inline content-src/selectors/selectors.js]: yeah, @bryanbell changed his mind :P

File: content-src/components/Spotlight/Spotlight.js (hunk 1 above)
Comment (k88hudson) [inline content-src/components/Spotlight/Spotlight.js]: Ah, interesting.

Comment (k88hudson) [inline content-src/components/Spotlight/Spotlight.js]: I will add a note in that ticket

Comment (pdehaan) [inline content-src/components/Spotlight/Spotlight.js]: Obviously a problem for another day (like Post V1 when we get to that issue). Unless we think it'd be confusing to show URLs on only 1/2 of the results.

**Comment by k88hudson** [comment]
mind taking a look at this @rlr?

File: content-src/selectors/selectors.js [hunk 3]
```diff
@@ -7,26 +7,59 @@ const urlParse = require("url-parse");
...
+function getBackgroundRGB(site) {
+  if (site.favicon_colors && site.favicon_colors[0] && site.favicon_colors[0].color) {
+    return site.favicon_colors[0].color;
+  }
+
+  const favicon = site.favicon_url || site.favicon;
+  const parsedUrl = site.parsedUrl || urlParse(site.url || "") ;
+  const label = prettyUrl(parsedUrl.hostname);
+  let backgroundRGB = favicon ? DEFAULT_FAVICON_BG_COLOR : getRandomColor(label);
+  try { backgroundRGB = site.favicon_colors[0].color || backgroundRGB; } catch (e) {/**/}
```
Comment (rlr) [inline content-src/selectors/selectors.js]: This line confuses me. Is it needed? We check for `site.favicon_colors[0].color` at the top and return early if it exists.

File: content-src/selectors/selectors.js [hunk 4]
```diff
@@ -7,26 +7,59 @@ const urlParse = require("url-parse");
...
+    .sort((site1, site2) => {
+      const one = isValidSpotlightSite(site1);
+      const two = isValidSpotlightSite(site2);
+      if (one && two) {
+        return 0;
+      } else if (two) {
+        return 1;
+      } else {
+        return -1;
+      }
```
Comment (rlr) [inline content-src/selectors/selectors.js]: took me a minute to figure out, but cool! :+1:

**Comment by rlr** [comment]
Awesome! I *think* that one line can be safely removed but I totally could be missing something. Other than that, R+

**Pull request merged**